packages/trpc/server/routers/viewer/apps/toggle.handler.ts

```diff
@@ -69,11 +69,31 @@ export const toggleHandler = async ({ input, ctx }: ToggleOptions) => {
         },
       });
 
-      // TODO: This should be done async probably using a queue.
-      Promise.all(
-        appCredentials.map(async (credential) => {
+      class Queue<T> {
+        private items: T[] = [];
+
+        enqueue(item: T) {
+          this.items.push(item);
+        }
+
+        dequeue(): T | undefined {
+          return this.items.shift();
+        }
+
+        isEmpty(): boolean {
+          return this.items.length === 0;
+        }
+      }
```

Comment:
Should we extract Queue so we can use it elsewhere in the code if required? Feels a bit weird just hardcoding this class here imho

Reply:
Thanks @sean-brydon for reviewing my PR. Yeah, I definitely agree with you. Will make that change, np. Thanks.

Reply:
Hey @sean-brydon would like to suggest where to extract the Queue class, or should I go ahead raising a PR and get it reviewed?